src/bridges/convex/ConvexStakingBridge.sol

```diff
@@ -0,0 +1,223 @@
+        uint256 _totalInputValue,
+        uint256,
+        uint64,
+        address _rollupBeneficiary
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 outputValueA,
+            uint256,
+            bool
+        )
+    {
+        if (_totalInputValue == 0) {
+            revert ErrorLib.InvalidInputAmount();
+        }
+
+        if (
+            _inputAssetA.assetType != AztecTypes.AztecAssetType.ERC20 ||
+            _outputAssetA.assetType != AztecTypes.AztecAssetType.ERC20
+        ) {
+            revert InvalidAssetType();
+        }
+
+        if (
+            deployedTokens[_inputAssetA.erc20Address] == address(0) &&
+            deployedTokens[_outputAssetA.erc20Address] == address(0)
+        ) {
+            revert UnknownAssetA(); // invalid address or pool has not been loaded yet/token not deployed yet  --> this will need to get tested
+        }
+
+        PoolInfo memory selectedPool;
+
+        if (deployedTokens[_inputAssetA.erc20Address] == _outputAssetA.erc20Address) {
+            selectedPool = pools[_inputAssetA.erc20Address];
+
+            uint256 startCurveRewards = ICurveRewards(selectedPool.curveRewards).balanceOf(address(this));
+
+            BOOSTER.deposit(selectedPool.poolId, _totalInputValue, true);
+
+            uint256 endCurveRewards = ICurveRewards(selectedPool.curveRewards).balanceOf(address(this));
+
+            outputValueA = (endCurveRewards - startCurveRewards);
+
+            IRepConvexToken(_outputAssetA.erc20Address).mint(_totalInputValue);
+        } else if (deployedTokens[_outputAssetA.erc20Address] == _inputAssetA.erc20Address) {
+            // withdrawal
+
+            selectedPool = pools[_outputAssetA.erc20Address];
+
+            uint256 startCurveLpTokens = ICurveLpToken(_outputAssetA.erc20Address).balanceOf(address(this));
+
+            // transfer CONVEX tokens from CrvRewards back to the bridge
+            ICurveRewards(selectedPool.curveRewards).withdraw(_totalInputValue, true); // always claim rewards
+
+            BOOSTER.withdraw(selectedPool.poolId, _totalInputValue);
+
+            uint256 endCurveLpTokens = ICurveLpToken(_outputAssetA.erc20Address).balanceOf(address(this));
+
+            outputValueA = (endCurveLpTokens - startCurveLpTokens);
+
+            IRepConvexToken(_inputAssetA.erc20Address).burn(_totalInputValue);
+        } else {
+            revert ErrorLib.InvalidOutputA();
+        }
+
+        // Pay out subsidy to the rollupBeneficiary
+        SUBSIDY.claimSubsidy(
+            computeCriteria(_inputAssetA, _inputAssetB, _outputAssetA, _outputAssetB, 0),
+            _rollupBeneficiary
+        );
+    }
+
+    /**
+    @notice Loads pool information for all pools supported by Convex Finance.
+    @notice Set allowance for Rollup's Curve LP tokens.
+    @notice Cached. Loads only new pools.
+    */
+    function loadPools() public {
```

Comment:
Given that we have to list all the tokens on RollupProcessor as well it makes sense to change this functions so that it takes `poolId` as a parameter and sets up only that one pool. Deploying contracts is expensive so it will save us a lot of money on gas on the pools we would not use. I propose having this function signature:
`function listPool(uint256 _poolId) external;`.

Reply:
Resolved